pkg/search/result.go

```diff
@@ -0,0 +1,143 @@
+package search
+
+import (
+	"reflect"
+	"strings"
+	"time"
+)
+
+var RepositoryFields = []string{
```

Comment:
Some of the repository field names differ from the ones listed in `gh repo list --json` and this could cause confusion:
- `archived` → `isArchived`
- `fork` → `isFork`
- `private` → `isPrivate`

Additionally, could more repository fields be exposed? The idea is that the JSON output should expose more information about repositories than that is included in the table output. For example, the list of fields in `gh repo list --json`:
```
  assignableUsers
  codeOfConduct
  contactLinks
  createdAt
  defaultBranchRef
  deleteBranchOnMerge
  description
  diskUsage
  forkCount
  fundingLinks
  hasIssuesEnabled
  hasProjectsEnabled
  hasWikiEnabled
  homepageUrl
  id
  isArchived
  isBlankIssuesEnabled
  isEmpty
  isFork
  isInOrganization
  isMirror
  isPrivate
  isSecurityPolicyEnabled
  isTemplate
  isUserConfigurationRepository
  issueTemplates
  issues
  labels
  languages
  latestRelease
  licenseInfo
  mentionableUsers
  mergeCommitAllowed
  milestones
  mirrorUrl
  name
  nameWithOwner
  openGraphImageUrl
  owner
  parent
  primaryLanguage
  projects
  pullRequestTemplates
  pullRequests
  pushedAt
  rebaseMergeAllowed
  repositoryTopics
  securityPolicyUrl
  squashMergeAllowed
  sshUrl
  stargazerCount
  templateRepository
  updatedAt
  url
  usesCustomOpenGraphImage
  viewerCanAdminister
  viewerDefaultCommitEmail
  viewerDefaultMergeMethod
  viewerHasStarred
  viewerPermission
  viewerPossibleCommitEmails
  viewerSubscription
  watchers
```
Not all of these are accessible via the REST API, but we could at least expose those that are.